source/Mission.cpp

```diff
@@ -855,6 +860,18 @@ const list<NPC> &Mission::NPCs() const
 
 
 
+// Update which NPCs are active based on their spawn and despawn conditions.
+void Mission::UpdateNPCs(const PlayerInfo &player)
+{
+	for(auto &npc : npcs)
+	{
+		npc.CanSpawn(player);
+		npc.CanDespawn(player);
```

Comment:
Should despawn conditions only be checked if the spawn conditions have succeeded? Right now an npc can fail its spawn conditions but succeed its despawn conditions. Then if the player's conditions later change such that the spawn conditions pass but the despawn conditions would no longer pass, the fleet can't spawn, as the despawn conditions has already succeeded before.